### DISCUSSION TITLE
Why did you use suppressHydrationWarning and react 18.3 canary?

### DISCUSSION THREAD

**Discussion by yannickpschroeder** [kiliman/remix-vite-mui#3] [Q&A] [answered]
It is not really an Issue, but there are no discussions activated for this repository.
I have 2 small questions:

Why is `suppressHydrationWarning` needed at https://github.com/kiliman/remix-vite-mui/blob/main/app/root.tsx#L23?
And do we really want to use the react 18.3 canary version, or would 18.2.0 be fine as well?

Thank you for your work!

**Comment by kiliman** [comment, marked as answer]
Remix hydrates the whole document instead of just a root `<div>`. This causes hydration issues when you have things like browser extensions that mutate the DOM (like `<head>`) before hydration. This is a problem in React 18.2. However, they fixed this in the latest Canary version. At the time of this example, it was 18.3, but now it is v19. 

NOTE: Next.js uses React 19 Canary, and the React team says it is fully supported.

As for the `suppressHydrationWarning`, that only applies to DOM attribute mismatches. Some extraneous `data-` attributes were being added by tooling, and I wanted to ensure the DevTools console was clean. This is a personal preference. This is not the same as the hydration warnings that v19 solves.

**Reply by yannickpschroeder** [reply]
Thank you very much for that quick answer!

**Reply by yannickpschroeder** [reply]
@kiliman Interestingly, version 18.3.0 was released today at https://www.npmjs.com/package/react. When I replace the 18.3.0-canary-01ab35a9a-20240228 version used in this repository with 18.3.0, I get hydration errors. Do you have any idea why this could be?

**Reply by kiliman** [reply]
Yes, because v18.3 is just some minor fixes for 18.2. So, they still have hydration issues.

The new Canary versions are v19.

**Reply by yannickpschroeder** [reply]
Thank you for the even quicker answer. Just recognized I had a misunderstanding about react canary Versions.
